packages/gestalt/src/behaviors/OutsideEventBehavior.js

```diff
@@ -1,39 +1,35 @@
 // @flow
-import * as React from 'react';
-import { findDOMNode } from 'react-dom';
+import type { Element } from 'react';
+import React, { useEffect } from 'react';
 
 type Props = {|
-  children: React.Node,
+  children: Element<*> | Array<Element<*>>,
   onClick?: (event: MouseEvent) => void,
 |};
 
-export default class OutsideEventBehavior extends React.Component<Props> {
-  componentDidMount() {
-    document.addEventListener('click', this.handleClickEvent, {
-      capture: true,
-    });
-  }
-
-  componentWillUnmount() {
-    document.removeEventListener('click', this.handleClickEvent, {
-      capture: true,
-    });
-  }
-
-  handleClickEvent = (event: MouseEvent) => {
-    // eslint-disable-next-line react/no-find-dom-node
-    const el = findDOMNode(this);
-    if (
-      !this.props.onClick ||
-      !el ||
-      (event.target instanceof Node && el.contains(event.target))
-    ) {
+const OutsideEventBehavior = ({ children, onClick }: Props) => {
+  let isClickedInside = false;
+  const handleClickEvent = (event: MouseEvent) => {
```

Comment:
@emilyalwayss can we rename this to `handleDocumentClickEvent`? Differentiates with the other `handleClick` we now have.